### PR TITLE
✨ feat: implementing remaining grammar features

### PR DESCRIPTION
**Pull request by rduo1009**
Supersedes #181.

Remaining grammar features in `accido` and `transfero`. Unfortunately with some breaking changes.
- [x] passives for all tenses
- [x] remaning subjunctive forms (passive subjunctives)
- [x] deponent verbs
- [x] gerunds 
- [x] gerundives (future passive participles)
- [x] supines
- [x] more imperatives
- [x] more infinitives
- [x] active only verbs
- [x] verbs that lack supine/ppp
- [x] verbs that are impersonal
- [x] verbs that are impersonal in passive
- [x] verbs with future active participles as fourth principal part
- [x] verbs that lack perfect
- [x] verbs that lack future
- [x] verbs that lack future active participle
- [x] improving edge cases (`TODO`s in [src/core/accido/edge_cases.py](https://github.com/rduo1009/vocab-tuister/blob/main/src/core/accido/edge_cases.py) and [src/core/transfero/edge_cases.py](https://github.com/rduo1009/vocab-tuister/blob/main/src/core/transfero/edge_cases.py))
- [x] 'facio' is irregular, with some derived variants as well (check for more like this)
- [x] perfect only verbs and no-infinitive verbs (do manually in `DEFECTIVE_VERBS`)
- [x] fixing bugs with irregular 'eo', etc. based verbs
- [x] more uses of imperfect
- [x] English inflections of subjunctive
- [x] semi-deponent verbs
- [x] ~~improving and then integrating synonyms into project (under misc setting)~~

## Summary by CodeRabbit

* **New Features**
  * Added granular exclusion options for deponent verbs, gerundives, gerunds, supines, and a wide range of verb tenses, moods, and voices in configuration and UI.
  * Introduced support for English subjunctive and verbal noun translations as configurable options.
  * Expanded handling and documentation for verbs lacking perfect passive participles and irregular noun declensions.
  * Enhanced English inflection generation to cover passive, subjunctive, and verbal noun forms.
  * Improved synonym search with part-of-speech filtering and inclusion of similar words.

* **Bug Fixes**
  * Corrected handling of verbs and participles missing certain forms, ensuring accurate form omission and error reporting.
  * Fixed and expanded test coverage for passive, subjunctive, deponent, and irregular verb and noun forms.

* **Documentation**
  * Updated and clarified documentation for verb and noun handling, configuration options, and AI agent guidelines.

* **Style**
  * Improved code formatting, indentation, and type annotations across multiple files.
  * Updated static type checking configurations to use Pyright.

* **Chores**
  * Added and updated configuration, test, and linting files to support new features and improved type checking.
  * Expanded and reorganized test data and settings for comprehensive coverage.

### CODE REVIEW

File: tests/rogo_test/typein_engtolat_test.py [hunk 1]
```diff
@@ -120,7 +131,7 @@ def test_typein_engtolat():
     for output in ask_question_without_sr(vocab_list, amount, settings):
         assert type(output) is TypeInEngToLatQuestion
 
-        ic(output)  # type: ignore[name-defined] # noqa: F821
+        ic(output)  # noqa: F821
```
[the remark this answers is not in the document]
Comment (rduo1009) [reply tests/rogo_test/typein_engtolat_test.py:134]: `ic` is imported by the conftest.py file before tests are ran.

**Comment by rduo1009** [comment]
This PR does not complete all of the core `vocab-tuister` features. Some more edge cases need to be added. Also integrating synonyms will be implemented later as well. I was also considering adding the ability to use multiple meanings in a `MultipleMeanings` to narrow down synonyms.
Anyway, improving test coverage is next.
